ffi/derive/src/convert.rs

```diff
@@ -20,140 +27,182 @@ pub fn derive_ffi_type(mut input: DeriveInput) -> TokenStream {
         Data::Enum(item) => {
             let repr = find_attr(&input.attrs, "repr");
 
-            if item.variants.is_empty() {
-                abort!(name, "uninhabited enum's cannot be instantiated");
-            }
-
             if is_fieldless_enum(&item) {
+                if item.variants.len() == 1 {
+                    // NOTE: one-variant fieldless enums have representation of ()
+                    return derive_ffi_type_for_opaque_item(name, &input.generics);
+                }
+                if without_repr(&input.attrs) {
+                    abort!(name, "Fieldless enum missing #[repr(int)]");
+                }
                 if item.variants.iter().any(|v| v.discriminant.is_some()) {
                     abort!(
                         name,
-                        "fieldless enums with explicit discriminants are prohibited"
+                        "Fieldless enums with explicit discriminants are prohibited"
                     );
                 }
+
                 derive_ffi_type_for_fieldless_enum(&input.ident, &item, &repr)
             } else {
                 let local = !is_non_local(&input.attrs);
                 derive_ffi_type_for_data_carrying_enum(&input.ident, input.generics, &item, local)
             }
         }
-        Data::Struct(_) => {
-            if is_owning(&input.attrs, &input.data) {
-                abort!(
-                    name,
-                    "Structure contains raw pointers. If the pointers don't own the data, attach `#[ffi_type(unsafe {non_owning})`"
-                )
-            }
+        Data::Struct(ref item) => {
```

Comment:
But you don't use `ref` in `Data::Enum(item)`...

Reply:
is that a problem?

Reply:
ok, made the change. But mind you it wasn't just about style. One arm was capturing by value others by reference

Reply:
You can't write a diverging thing like that without the `ref` capture in the match arm